src/js/store/orgs/actions.ts

```diff
@@ -348,3 +353,37 @@ export const commitFailed = ({
     payload: model,
   });
 };
+
+export const refreshOrg = (org: Org): ThunkResult => async (dispatch) => {
+  const url = window.api_urls.scratch_org_refresh(org.id);
+  dispatch({
+    type: 'REFRESH_ORG_STARTED',
+    payload: { org, url },
+  });
+  try {
+    const response = await apiFetch({
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      url: addUrlParams(url),
+      dispatch,
+      opts: {
+        method: 'POST',
+        body: JSON.stringify(org),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      },
+    });
+    if (!response) {
+      return dispatch({
+        type: 'REFRESH_ORG_FAILED',
+        payload: { org, url, response },
+      });
+    }
+    return dispatch({
+      type: 'REFRESH_ORG_SUCCEEDED',
+      payload: { org: response, url },
+    });
+  } catch (err) {
+    console.log(err);
+  }
+};
```

Comment:
We'll also need actions here that _respond_ to the back-end WS events, indicating whether the refresh eventually succeeded or failed.

In the case of success, we can show a toast and then just call `updateOrg` with the org to update it in the front end store. In the case of error, same thing but with an error toast. I think both actions can be very similar to this one: https://github.com/oddbird/MetaShare/pull/162/files#diff-d69236bdfc813971102a3ad7cdf2bbb9R114-R131

But, we also need a way for those actions to be called when we receive the WS events from the back end. In this case, that means adding some handling in `websockets.ts`, to receive the events and call the right actions. I think you can just copy how it's handling `SCRATCH_ORG_PROVISION` and `SCRATCH_ORG_PROVISION_FAILED`, but with `SCRATCH_ORG_REFRESH` and `SCRATCH_ORG_REFRESH_FAILED` instead, and calling the new actions mentioned above in this comment.

Reply:
I think this is the part I'm missing.  On `success` I'm calling `updateOrg` instead of creating a new event that displays the success toast.